src/tbp/monty/frameworks/config_utils/make_dataset_configs.py

```diff
@@ -1102,3 +1104,279 @@ class FiveLMMountConfig:
+            sorting_inds = np.argsort(angles)
+            inds = inds[sorting_inds]
+            indices.extend([row for row in inds])
+
+    elif order_by == "spiral":
+        indices = [(i_mid, i_mid)]
+
+        # Directions for moving in spiral: right, down, left, up
+        directions = [(0, 1), (1, 0), (0, -1), (-1, 0)]
+        current_dir = 0  # Start moving right
+        steps = 1  # How many steps to take in current direction
+        steps_taken = 0  # Steps taken in current direction
+        row, col = i_mid, i_mid  # Start at center
+
+        # Generate spiral pattern until we have enough points
+        while len(indices) < grid_size**2:
+            # Move in current direction
+            row += directions[current_dir][0]
+            col += directions[current_dir][1]
+
+            # Add point if it's within bounds
+            if 0 <= row < grid_size and 0 <= col < grid_size:
+                indices.append((row, col))
+
+            steps_taken += 1
+
+            # Check if we need to change direction
+            if steps_taken == steps:
+                steps_taken = 0
+                current_dir = (current_dir + 1) % 4
+                # Increase steps every 2 direction changes (completing half circle)
+                if current_dir % 2 == 0:
+                    steps += 1
+
+    indices = np.array(indices)[:n_sensors]
+
+    # Convert indices to locations in agent space.
+    positions = []
+    for idx in indices:
+        positions.append((x[idx[0], idx[1]] * delta, y[idx[0], idx[1]] * delta))
+    positions = np.array(positions)
+
+    # Add z-positions.
+    positions = np.hstack((positions, np.zeros([positions.shape[0], 1])))
+
+    # Optionally append entry for a view finder which is a duplicate of row zero.
+    # Should be (0, 0, 0).
+    if add_view_finder:
+        positions = np.vstack([positions, positions[0].reshape(1, -1)])
+
+    return positions
+
+
+def make_multi_sensor_mount_config(
+    n_sensors: int,
+    agent_id: str = "agent_id_0",
+    sensor_ids: Optional[Sequence[str]] = None,
+    height: Number = 0.0,
+    position: ArrayLike = (0, 1.5, 0.2),  # agent position
+    resolutions: Optional[ArrayLike] = None,
+    positions: Optional[ArrayLike] = None,
+    rotations: Optional[ArrayLike] = None,
+    semantics: Optional[ArrayLike] = None,
+    zooms: Optional[ArrayLike] = None,
+) -> Mapping[str, Any]:
+    """Generate a multi-sensor mount configuration.
+
+    Creates a multi-sensor, single-agent mount config. Its primary use is in generating
+    a `MultiLMMountHabitatDatasetArgs` config. This function assumes that a view finder
+    will be included which entails that the number of sensor_ids, resolution, etc. will
+    be of length `n_sensors + 1`. Defaults are reasonable and reflect current common
+    practices.
+
+    Args:
+        n_sensors: Number of sensors. Count should not include an optional view finder.
```

Comment:
This suggests the viewfinder is option, but further below `add_view_finder=True` appears to be hard-coded, is it worth passing in a parameter for this? Similarly this should presumably influence the integer value of `arr_len`?

Alternatively maybe just don't state that the view-finder is optional.

Reply:
Yeah, I toyed with making the view finder optional throughout but it got complicated. I think I'd like to keep the `add_view_finder` argument in `make_sensor_positions_on_grid` in case we have a use for it in the future and it's not hurting anything to leave with `True` as the default. I've updated the docstring though to clarify the relationship between `n_sensors` and the view finder.